velox/type/TimestampConversion.cpp

```diff
@@ -218,7 +218,9 @@ bool tryParseDateString(
 
   // No month or day.
   if (mode == ParseMode::kNonStandardCast && pos == len) {
-    daysSinceEpoch = daysSinceEpochFromDate(year, 1, 1);
+    if (!daysSinceEpochFromDate(year, 1, 1, daysSinceEpoch).ok()) {
+      return false;
+    }
```

Comment:
Maybe simplified with VELOX_RETURN_IF?

Reply:
Looks like it cannot. VELOX_RETURN_IF returns status code.